### ISSUE TITLE
Specifying http proxy changed in Node@18 and 20

### ISSUE THREAD

**Issue by waldekmastykarz** [microsoft/kiota-typescript#1103]
Since v18, Node ships with a native fetch client. This client requires a different way to specify an HTTP proxy than what we've been supporting in our SDKs so far.

To specify an HTTP proxy for Node's fetch, you need to use the following code (tested on Node v18.19 and v20.11):

```ts
import { ProxyAgent } from 'undici';
import { Client } from '@microsoft/microsoft-graph-client';

const dispatcher = new ProxyAgent(process.env.https_proxy);
const fetchOptions = {
  dispatcher
};

Client.initWithMiddleware({
  middleware,
  fetchOptions
});
```

What's noteworthy:
- we specify HTTP proxy using dispatcher instead of agent (agent is ignored on requests, which makes it hard to debug)
- the ProxyAgent dispatcher comes from undici which seems to be maintained by Node folks. The ProxyAgent is incompatible with other agents that we used to use in the past.

We should update our FetchOptions interface to support specifying dispatcher.

**Comment by baywet** [comment]
Thanks for reaching out.

@sebastienlevert and I were chatting about that earlier and he mentioned he made it work. I'll let him provide the specifics when he has time (as you know I'm traveling right now and don't have time to dig into this)

**Comment by sebastienlevert** [comment]
Just to confirm... This is for Graph JS v3, right? We donc have the middleware initialization anymore IIRC.

As we won't be adding new features over there, I highly doubt we'll get to this.

**Comment by waldekmastykarz** [comment]
This change affects any code that relies on Node's fetch. I noticed it in Graph JS SDK v3 but if the new TypeScript SDK also uses Node.js fetch, then it would also have the same issue.

**Comment by waldekmastykarz** [comment]
Yes it does! I appreciate you being mindful of this requirement, that many large organization need for their apps.

**Comment by baywet** [comment]
I have meant to come back to this issue for over a month now but failed the time to do so.
@rkodev when you have couple of minutes, can you please look into this?
